liquidctl/util.py

```diff
@@ -404,6 +406,117 @@ def map_direction(direction, forward=None, backward=None):
         raise ValueError(f'invalid direction: {direction!r}')
 
 
+def fan_mode_parser(value: Optional[str], max_fans: Optional[int] = None) -> dict:
```

Comment:
After merging this, we should look into adding more type annotations, and checking them (again).

IIRC one issue is our use of:

```py
class Base:
    def some_method(arg1, arg2, **kwargs):
        raise NotImplementedError()
```

and each subclass having their implementations _with slightly different signatures:_

```py
def some_method(arg1, arg2, arg3=None, arg4=None, **kwargs):
    # ...
```

(If we end up deciding to continue _not_ using type annotations, they should then be removed from this function).